src/material-experimental/dialog/dialog-container.ts

```diff
@@ -0,0 +1,217 @@
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ElementRef,
+  HostBinding,
+  ViewChild,
+  ComponentRef,
+  EmbeddedViewRef,
+  ChangeDetectorRef,
+  Component,
+  Optional,
+  Inject,
+  ViewEncapsulation,
+  ChangeDetectionStrategy,
+} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+import {trigger, state, style, transition, animate, AnimationEvent} from '@angular/animations';
+import {
+  BasePortalHost,
+  PortalHostDirective,
+  ComponentPortal,
+  TemplatePortal
+} from '@angular/cdk/portal';
+import {FocusTrapFactory} from '@angular/cdk/a11y';
+import {DialogConfig} from './dialog-config';
+import {Subject} from 'rxjs/Subject';
+
+
+export function throwDialogContentAlreadyAttachedError() {
+  throw Error('Attempting to attach dialog content after content is already attached');
+}
+
+
+/**
+ * Internal component that wraps user-provided dialog content.
+ * @docs-private
+ */
+@Component({
+  moduleId: module.id,
+  selector: 'cdk-dialog-container',
+  template: `<ng-template cdkPortalOutlet></ng-template>`,
+  styles: [`
+    cdk-dialog-container {
+      background: white;
+      border-radius: 5px;
+      display: block;
+      padding: 10px;
+    }
+  `],
+  encapsulation: ViewEncapsulation.None,
+  preserveWhitespaces: false,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [
+    trigger('dialog', [
+      state('enter', style({ opacity: 1 })),
+      state('exit, void', style({ opacity: 0 })),
+      transition('* => *', animate(225)),
+    ])
+  ],
+  host: {
+    '[@dialog]': '_state',
+    '(@dialog.start)': '_onAnimationStart($event)',
+    '(@dialog.done)': '_onAnimationDone($event)',
+  },
+})
+export class CdkDialogContainer extends BasePortalHost {
+  /** State of the dialog animation. */
+  _state: 'void' | 'enter' | 'exit' = 'enter';
+
+  /** Element that was focused before the dialog was opened. Save this to restore upon close. */
+  private _elementFocusedBeforeDialogWasOpened: HTMLElement | null = null;
+
+   /** The class that traps and manages focus within the dialog. */
+  private _focusTrap = this._focusTrapFactory.create(this._elementRef.nativeElement, false);
+
+  // tslint:disable-next-line:no-host-decorator-in-concrete
+  @HostBinding('attr.aria-label') private get _ariaLabel() {
+    return this._config.ariaLabel || null;
+  }
+
+  // tslint:disable-next-line:no-host-decorator-in-concrete
+  @HostBinding('attr.aria-describedby')
+  private get ariaDescribedBy() { return this._config ? this._config.ariaDescribedBy : null; }
+
+  // tslint:disable-next-line:no-host-decorator-in-concrete
+  @HostBinding('attr.role') private get role() { return this._config ? this._config.role : null; }
+
+  // tslint:disable-next-line:no-host-decorator-in-concrete
+  @HostBinding('attr.tabindex') private get tabindex() { return -1; }
```

Comment:
Didn't we have a rule to use the `host` instead of `@HostBinding`?

Reply:
The intention of `CdkDialogContainer` is to be extended so we use `@HostBinding` to allow for it to be inherited.